contracts/contracts/core/BlockTracker.sol

```diff
@@ -89,6 +90,12 @@ contract BlockTracker is IBlockTracker, BlockTrackerStorage,
         emit NewL1Block(_blockNumber, _winner, currentWindow);
     }
 
+    /// @dev Allows the owner to set the provider registry.
+    function setProviderRegistry(address newProviderRegistry) external onlyOwner {
+        providerRegistry = IProviderRegistry(newProviderRegistry);
+        emit ProviderRegistrySet(address(providerRegistry), newProviderRegistry);
```

Comment:
Doesn’t this emit the event with the same two variables?

Reply:
Looks like this still needs to be fixed